src/CodeImport/DoItChunk.class.st

```diff
@@ -149,8 +149,11 @@ DoItChunk >> buildClassFromOldDefinition: ast [
 	(#( CompiledBlock CompiledCode CompiledMethod ) includes: subclassName) ifTrue: [ layoutClass := CompiledMethodLayout ].
 
 	class := self class classInstaller make: [ :aBuilder |
+			| superclass |
+			superclass := self class environment at: superclassName asSymbol.
```

Comment:
There is something strange that can break the UndefinedClass because the environment of the Chunk may not be the correct one. May be the classBuilder is working on another one. 
If we find a superclass in the environment of the DoItChunk then may be this is the wrong one. 
In any case I think that this is important to pass the name if we do not find the superclass because like that the builder can use the UndefinedClass as a superclass.